core/src/main/scala/com/eharmony/spotz/optimizer/StopStrategy.scala

```diff
@@ -0,0 +1,85 @@
+package com.eharmony.spotz.optimizer
+
+import org.joda.time.Duration
+
+/**
+  * @author vsuthichai
+  */
+sealed trait StopStrategy extends Serializable {
+  final val UNLIMITED = Long.MaxValue
+  final val FOREVER = new Duration(Long.MaxValue)
+
+  def getMaxTrials: Long = UNLIMITED
```

Comment:
Relic of the past that wasn't deleted when I refactored the trait.  Originally, the two main requirements for stopping criteria were that the user could specify maximum number of trials and/or time duration.  It has since grown to support other things.